### PR TITLE
PE-2272: limit private files download size on mobile

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
## Overview
Add a size limit to download a private file. 

### Refactors and new entities
We are adding new ARFS entities, inspired by the [ardrive-core-js](https://github.com/ardriveapp/ardrive-core-js). This will be a continuous task and we'll add the entities as we are touching them.

This PR adds the following ones:
`ARFSEntity`
`ARFSPrivateEntity`
`ARFSFileEntity`
`ARFSPrivateFileEntity`
`ARFSDriveEntity`

We are adding a factory - that can be an Adapter - `ARFSFactory` please feel free to add a comment about that. This factory will create the correct object from any source e.g. Database, Graphql, etc...

### ProfileFileDownloadCubit
We also refactored the `ProfileFileDownloadCubit` and was added tests for that.

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/3r7tp0fdqe7v8
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/7aqf3387rv08o

### CODE REVIEW

File: lib/blocs/file_download/personal_file_download_cubit.dart [hunk 1]
```diff
@@ -10,72 +10,89 @@ class FileDownloadProgress extends LinearProgress {
...
     try {
-      final drive = await _driveDao.driveById(driveId: driveId).getSingle();
+      final drive = await _arfsRepository.getDriveById(_file.driveId);
+      final platform = SystemPlatform.platform;
 
-      switch (drive.privacy) {
+      switch (drive.drivePrivacy) {
         case DrivePrivacy.private:
-          await _downloadFile(fileName, fileSize, drive);
+          if (platform == 'Android' || platform == 'iOS') {
```
[the remark this answers is not in the document]
Comment (thiagocarvalhodev) [reply lib/blocs/file_download/personal_file_download_cubit.dart]: [32360e1](https://github.com/ardriveapp/ardrive-web/pull/750/commits/32360e15156763da71973a2d8ce97537f0500872)

**Review by karlprieb — DISMISSED** [review]

File: lib/l10n/app_en.arb [hunk 2]
```diff
@@ -669,6 +669,14 @@
   "@fileFailedToDownload": {
     "description": "Error message for failed downloads"
   },
+  "waitForDownload": "Due to the size of your file, this download could take a while. Please keep the app open while the download is in progress.",
```
Comment (matibat) [inline lib/l10n/app_en.arb:672]: Do we have tasks for dealing with translations for these new wordings?

Comment (matibat) [reply lib/l10n/app_en.arb:672]: Nit: in the future we may use a download notification so we can download in background

File: lib/utils/app_platform.dart [hunk 3]
```diff
@@ -1,36 +1,45 @@
...
-    case 'android':
-      return 'Android';
-    case 'ios':
-      return 'iOS';
-    default:
-      return 'unknown';
+  @visibleForTesting
+  static void setMockPlatform({required SystemPlatform platform}) {
+    _mockPlatform = platform;
   }
```
Comment (matibat) [inline lib/utils/app_platform.dart:13]: Great! I din't know there was a decoration for this, I'll start making use of it 🤩

File: lib/utils/app_platform.dart [hunk 4]
```diff
@@ -1,36 +1,45 @@
...
+        return SystemPlatform.unknown;
+    }
   }
+
+  static bool get isMobile =>
+      getPlatform() == SystemPlatform.Android ||
+      getPlatform() == SystemPlatform.iOS;
 }
+
+enum SystemPlatform { Android, iOS, Web, unknown }
```
Comment (matibat) [inline lib/utils/app_platform.dart:45]: Thank you 😎

**Review by matibat — APPROVED** [review]
LGTM

**Review by karlprieb — DISMISSED** [review]

**Review by matibat — DISMISSED** [review]

**Review by matibat — APPROVED** [review]

**Pull request merged**